src/addressResolvers/cache.ts

```diff
@@ -0,0 +1,40 @@
+import { Address, Handle } from './utils';
+import redis from '../helpers/redis';
+
+const KEY = 'address-resolvers';
+const TTL = 24 * 60 * 60 * 1e3;
+
+export async function getCache(addresses: Address[]): Promise<Record<Address, Handle>> {
+  const transaction = redis.multi();
+  addresses.map(address => {
+    transaction.get(`${KEY}:${address}`);
+  });
+  const results = await transaction.exec();
+
+  return Object.fromEntries(
+    addresses
+      .map((address, index) => {
+        return [address, results[index]];
+      })
+      .filter(([, handle]) => handle !== null)
+  );
+}
+
+export function setCache(payload: Record<Address, Handle>) {
+  const transaction = redis.multi();
+  Object.entries(payload).map(([address, handle]) => {
+    transaction.set(`${KEY}:${address}`, handle, { EX: TTL });
+  });
+
+  return transaction.exec();
+}
+
+export default async function cache(addresses: Address[], callback) {
+  const cache = await getCache(addresses);
+  const cachedAddresses = Object.keys(cache);
+
+  const results = await callback(addresses.filter(a => !cachedAddresses.includes(a)));
```

Comment:
we never use callbacks 🙈

Reply:
We do, in requestDeduplicator

Reply:
requestDeduplicator is different case but i haven't seen us using callbacks for normal usecase

Reply:
How will you accomplish this without passing a function to the `cache` function, and without coupling the cache logic inside addressResolvers ?

And how is requestDeduplicator different ? We pass a function to it, as well as the expected args to pass to the function, and return the results depending on the function result.